Tests/RxSwiftTests/PrimitiveSequenceTest.swift

```diff
@@ -1046,6 +1046,131 @@ extension PrimitiveSequenceTest {
 }
 
 extension PrimitiveSequenceTest {
+    func testFirst_Empty() {
+        let scheduler = TestScheduler(initialClock: 0)
+
+        let xs = scheduler.createHotObservable([
+            next(150, 1),
+            completed(250),
+            error(260, testError)
+            ])
+
+        let res: TestableObserver<Int> = scheduler.start { () -> Observable<Int> in
+            let single: Single<Int> = xs.first().map { $0 ?? -1 }
```

Comment:
I think you might take a look at how we've created
```
public func XCTAssertEqual<T: Equatable>(_ lhs: [Recorded<Event<T>>], _ rhs: [Recorded<Event<T>>], file: 
```
extensions and add
```
public func XCTAssertEqual<T: Equatable>(_ lhs: [Recorded<Event<T?>>], _ rhs: [Recorded<Event<T?>>], file: 
```
to `XCTest+AllTests.swift`.